### PR TITLE
IRO-1403 - Show Miner's Fee

### PR DESCRIPTION
**Pull request by brekk**
## Summary

Large numbers are hard to parse. There was an existing function which reformatted raw fees, but it seemed to either be broken / partially working.

I refactored the existing function `getNumberToUnit` in order to account for negative numbers, so now "-2010000000" becomes "-20.10 $IRON".

I also added a visual indicator for mining fees when the fee is negative. It looks like this:

CleanShot 2022-02-11 at 13 48 34

On small breakpoints it looks like this:

CleanShot 2022-02-11 at 14 31 52

## Testing Plan

Browse around and see whether the pages look right / are more intuitive now.

## Breaking Change

No

### CODE REVIEW

File: src/types/Transaction.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { Spends } from './Spend'
 interface ParsedTransaction {
   id: number
   hash: string
-  fee?: number
+  fee?: string
```
Comment (brekk) [inline src/types/Transaction.ts:8]: All of the data I saw suggested that transaction fees are already strings. It looks like there may be a pattern for converting fees from string to number on parse, but relative to what was already in place (and the fact that downstream functions were consuming strings but pretending they were numbers).

File: src/utils/currency.test.ts [hunk 2]
```diff
@@ -1,26 +1,66 @@
...
+    fixture.map(([input, output]) => {
+      const int = parseInt(input)
+      const i = getIRFAmountWithCurrency(input)
+      expect(i).toEqual(output)
+      if (int) {
+        expect(getIRFAmountWithCurrency('-' + input)).toEqual('-' + output)
+      }
+    })
+  })
+  test('getNumberToUnit', () => {
```
Comment (brekk) [inline src/utils/currency.test.ts:37]: See earlier commits for verification that this refactored `getNumberToUnit` works as expected (only the new one can deal with negative values).

**Review by dguenther — APPROVED** [review]
Nice, looks good! Nothing blocking here, just some commented code that would be nice to get rid of before this merges. I'll leave an approve on it and feel free to merge when you're ready.

File: src/components/TransactionsList/TransactionsList.tsx [hunk 3]
```diff
@@ -53,25 +53,37 @@ const TransactionsList = (props: Prop) => {
...
-                <StyledTableCell align='right'>
-                  {getIRFAmountWithCurrency(transaction.fee)}
-                </StyledTableCell>
-                <StyledTableCell align='right'>
-                  {getDisplaySizeInBytes(transaction.size)}
-                </StyledTableCell>
-              </StyledTableRow>
-            ))}
+            {transactions.map(({ fee = '0', hash, size }) => {
+              const formattedFee = parseInt(fee).toLocaleString()
```
[the remark this answers is not in the document]
Comment (brekk) [reply src/components/TransactionsList/TransactionsList.tsx:57]: I changed this one to use BigInt.

File: src/utils/currency.ts [hunk 4]
```diff
@@ -1,18 +1,22 @@
-const ORE_TO_IRON = 100000000
+export const ORE_TO_IRON = 1e8
 const ORE_TICKER = '$ORE'
 const IRON_TICKER = '$IRON'
 
-export const getIRFAmountWithCurrency = (amount: number | undefined): string => {
+export const getIRFAmountWithCurrency = (raw = '0'): string => {
+  const amount = parseInt(raw)
```
[the remark this answers is not in the document]
Comment (brekk) [reply src/utils/currency.ts:6]: Relative to this one, hitting both:
- `2345: Argument of type 'bigint' is not assignable to parameter of type 'number'.` when trying to do `Math.abs`
- AFAICT you can't use the `/` operator with BigInts? I'm seeing: `2365: Operator '/' cannot be applied to types 'bigint' and 'number'.` -- I could ostensibly change a bunch of numbers to be bigints, but it seems more strange to do that?

So I'm thinking I'll leave these two functions as is.

**Review by deekerno — APPROVED** [review]
Looks good to me; I'm also not sure why the types on fee changed, but that's something we can address in more detail in a separate PR.

File: src/utils/currency.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (dguenther) [reply src/utils/currency.ts:6]: For the first, that's not a significant issue, `abs` is pretty easy to do manually. For the second, you can use division with bigints, you just need both numbers to be bigints.

It's fine to leave it for another time since it's not likely to be an issue in the near future, and I think our API is storing the column as a 32-bit integer right now. [But the field is specifically a bigint](https://github.com/iron-fish/ironfish/blob/3ed863022ceb6cdb76868e9c35c24d2efa916b24/ironfish/src/primitives/transaction.ts#L158) in the node code, along with values, so this is a mismatched type here.

File: src/utils/currency.test.ts [hunk 5]
```diff
@@ -1,26 +1,66 @@
...
+    // oh TS how I loathe you
+    const fixture: [number, string][] = [
+      [0, '0'],
+      [1, '1'],
+      [11, '11'],
+      [123, '123'],
+      [1234, '1.23'],
```
[the remark this answers is not in the document]
Comment (dguenther) [reply src/utils/currency.test.ts:43]: Yes, `getIRFAmountWithCurrency` doesn't call `getNumberToUnit` if the number is in the ~1000 range. But this function specifically is buggy, no? There's a test asserting that it converts `1234` to `1.23`, which is behavior that doesn't make sense for that range. It potentially could cause issues for whoever next has to modify this code or thinks it might be safe to use the function.
